auth/src/main/resources/api.yaml

```diff
@@ -0,0 +1,76 @@
+openapi: 3.0.1
+info:
+  description: 'Feast Authorization Server'
+  license:
+    name: Apache 2.0
+    url: http://www.apache.org/licenses/LICENSE-2.0.html
+  title: Feast Authorization Server
+  version: 1.0.0
+servers:
+  - url: /v1
+paths:
+  /healthz:
+    get:
+      responses:
+        "200":
+          description: Online
+        "500":
+          description: Offline
+  /readiness:
+    get:
+      responses:
+        "200":
+          description: Ready
+        "500":
+          description: Not Ready
+  /checkProjectAccess:
```

Comment:
it might be better to keep this API more generic to resources and separate the authorization decision from the ACL retrieval.  What we are finding in our implementation (to be shared shortly), is that the authorization decision can be generalized for different ACL providers (Keto, homegrown, etc).  In fact our initial implementation is to just create an adapter between our homegrown system and the Keto API.

I see value in pluggable authorization logic as well but I'd rather not take on that scope until there is a concrete need.  So my preference would be for:
`/principals/{principal}/acls?resourceType=project&resourceId=[projectId}`
but if we wanted to take on the auth decision then at least:
`/checkResourceAccess`

Reply:
I considered generalizing instead of creating a very specific endpoint, but I felt like it would increase the scope of the PR significantly. Also, nothing prevents us from adding more generalized endpoints over time.

In terms of taking on the authorization decision: Part of the reason that is within Feast is because Keto doesn't make that easy. Keto has a pretty low level API which requires pre- and post- processing of results in order to determine whether a user is authorized to access a resource. Many other solutions are more full fledged. 

What would be the response object in the first `GET` request? And how would that be used?

`checkResourceAccess` should be pretty straightforward to add, I'm assuming `resourceId` and `resourceType` would make its way into the query?